src/control/positionControl/pathPlanning/VoronoiPathPlanning.cpp

```diff
@@ -0,0 +1,153 @@
+//
+// Created by ratoone on 05-11-19.
+//
+
+#include "control/positionControl/pathPlanning/VoronoiPathPlanning.h"
+
+namespace rtt::ai::control{
+
+VoronoiPathPlanning::VoronoiPathPlanning(CollisionDetector& collisionDetector) : collisionDetector(collisionDetector){}
+
+void VoronoiPathPlanning::computeDiagram(const rtt::Vector2 &robotPosition, const rtt::Vector2 &targetPosition) {
+    auto robots = collisionDetector.getRobotPositions();
+    std::vector<jcv_point> robotPositions(robots.size());
+    std::transform(robots.begin(), robots.end(), robotPositions.begin(),
+            [](auto robot)-> jcv_point {return {(float)robot->x, (float)robot->y};});
+    jcv_rect playArea = {
+            (jcv_point){(float)std::min(robotPosition.x,targetPosition.x),(float)std::min(robotPosition.y,targetPosition.y)},
+            (jcv_point){(float)std::max(robotPosition.x,targetPosition.x),(float)std::max(robotPosition.y,targetPosition.y)}
+    };
+    jcv_diagram_generate(robots.size(), robotPositions.data(), &playArea, nullptr, &voronoiDiagram);
+}
+
+void VoronoiPathPlanning::generateGraphFromDiagram(){
+    graphAdjacencyList.clear();
+    for(const jcv_edge* edgeIterator = jcv_diagram_get_edges(&voronoiDiagram); edgeIterator != nullptr; edgeIterator = edgeIterator->next){
+        //ignore edges of the diagram
+        if (edgeIterator->a == 0 && edgeIterator->b == 0 && edgeIterator->c == 0){
+            continue;
+        }
+
+        double distance = rtt::Vector2(
+                edgeIterator->pos[0].x-edgeIterator->pos[1].x,
+                edgeIterator->pos[0].y-edgeIterator->pos[1].y).length();
+
+        //ignore degenerate lines
+        if (distance == 0){
+            continue;
+        }
+
+        rtt::Vector2 firstEdgePoint = convertFromJcvPoint(edgeIterator->pos[0]);
+        rtt::Vector2 secondEdgePoint = convertFromJcvPoint(edgeIterator->pos[1]);
+
+        //if the vertex is a new one (not found in the map), add it
+        if(graphAdjacencyList.empty() || graphAdjacencyList.find(firstEdgePoint) == graphAdjacencyList.end()){
+            graphAdjacencyList.insert({firstEdgePoint, std::list<GraphNode>(1,(GraphNode){secondEdgePoint, distance})});
+        }
+        //else append the adjacency list with the new neighbouring site and the new node connection
+        else{
+            graphAdjacencyList[firstEdgePoint].push_back((GraphNode){secondEdgePoint, distance});
+        }
+
+        //repeat for the second node
+        if(graphAdjacencyList.empty() || graphAdjacencyList.find(secondEdgePoint) == graphAdjacencyList.end()){
+            graphAdjacencyList[secondEdgePoint] = std::list<GraphNode>(1,(GraphNode){firstEdgePoint, distance});
+        }
+        else{
+            graphAdjacencyList[secondEdgePoint].push_back((GraphNode){firstEdgePoint, distance});
+        }
+    }
+}
+
+void VoronoiPathPlanning::completeGraphWithOriginDestination(const rtt::Vector2 &robotPosition, const rtt::Vector2 &targetPosition) {
```

Comment:
This whole function does almost exactly the same thing with robotPosition as it does with targetPosition. Consider making them one input and calling the function twice.